odoo/addons/base/models/ir_model.py

```diff
@@ -1819,44 +1810,45 @@ def unlink(self):
                 table = data.model.model.replace('.', '_')
 
             # double-check we are really going to delete all the owners of this schema element
-            self._cr.execute("""SELECT id from ir_model_constraint where name=%s""", [name])
-            external_ids = set(x[0] for x in self._cr.fetchall())
+            external_ids = {
+                id_ for [id_] in self.env.execute_query(SQL(
+                    """SELECT id from ir_model_constraint where name=%s""", name
+                ))
+            }
```

Comment:
It is tempting to refactor the code but it doesn't seems useful here ?

Reply:
I had to go through the whole class, so I replaced that usage of `_cr` too.